src/elmish-debugger/debugger.fs

```diff
@@ -5,30 +5,47 @@ open Fable.Core
 
 [<RequireQualifiedAccess>]
 module Debugger =
+    open FSharp.Reflection
+
+    let inline private duName (x:'a) = 
+        match FSharpValue.GetUnionFields(x, typeof<'a>) with
+        | case, _ -> case.Name
+
+    let [<PassGenericsAttribute>] inline private getCase cmd : obj =
```

Comment:
inlined functions are like compiler-approved macros, the body (code that *uses* generic information) is embedded at the callsite. At that point it's up to that method to have the type info, if the type is locally determined - all is well, if the type comes from elsewhere we recurse (attribute/inline), until we're all the way up the call chain.